### PR TITLE
Navigation

### PR DESCRIPTION
**Pull request by niquerio**

### CODE REVIEW

File: config/navigation.json [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      {
+        "title": "Interlibrary Loan"
+      },
+      {
+        "title": "Special Collections"
+      }
+    ]
+  },
+  {
+    "title": "Fines & fees",
```
Comment (erinesullivan) [inline config/navigation.json]:
```suggestion
    "title": "Fines and Fees",
```

File: config/navigation.json [hunk 2]
```diff
@@ -0,0 +1,74 @@
+[
+  { 
+    "title": "Account overview"
```
Comment (erinesullivan) [inline config/navigation.json]:
```suggestion
    "title": "Account Overview"
```

File: models/navigation.rb [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  def path
+    if parent?
+      "/#{@parent.slug}/#{slug}"
+    else
+      "/#{slug}"
+    end
+  end
+ 
+  def slug
+    if @title == 'Account overview'
```
Comment (erinesullivan) [inline models/navigation.rb]:
```suggestion
    if @title == 'Account Overview'
```

File: models/navigation.rb [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      ''
+    else
+      @title.gsub('/','or').gsub('&','and').gsub(/[\s]/,'-').downcase
+    end
+  end
+  def active?
+    if @title == 'Account overview'
```
Comment (erinesullivan) [inline models/navigation.rb]:
```suggestion
    if @title == 'Account Overview'
```

**Comment by jonearley** [comment]
This code looks great! The configuration file makes a lot of sense and the logic is clear. Thanks, @niquerio!

**Review by jonearley — APPROVED** [review]

**Review by erinesullivan — CHANGES_REQUESTED** [review]
Nit-picky titles.

File: views/navigation.erb [hunk 5]
```diff
@@ -29,14 +29,15 @@
...
   <div class="sticky-top">
     <ul class="site-navigation-list">
-      <% pages.each do |page| %>
-        <li class="<%= page[:to] == request.path_info ? 'active': '' %>">
-          <a href="<%=page[:to]%>">
-            <%=page[:label]%>
+      <% my_pages.pages.each do |page| %>
+        <li class="<%= page.active? ? 'active': '' %>">
+          <a href="<%=page.path%>">
+            <%=page.title%>
```
Comment (erinesullivan) [inline views/navigation.erb]:
```suggestion
            <m-icon name="<%=page.icon_name%>"></m-icon> <%=page.title%>
```

**Review by erinesullivan — CHANGES_REQUESTED** [review]
Adding icons to the sidebar navigation.

File: views/navigation.erb (hunk 5 above)
Comment (jonearley) [reply views/navigation.erb]: Hah! I was wondering about doing this too.

File: spec/models/navigation_spec.rb [hunk 6]
```diff
@@ -0,0 +1,120 @@
+describe Navigation do
+  it "to have pages" do
+    expect(described_class.new.pages.count).to be > 0
+  end
+  context "#cards" do
+    subject do
+      described_class.new.cards
+    end
+    it "should not have account overview listed" do
+      expect(subject.find{|x| x.title == 'Account overview'}).to be_nil
```
Comment (erinesullivan) [inline spec/models/navigation_spec.rb]:
```suggestion
      expect(subject.find{|x| x.title == 'Account Overview'}).to be_nil
```

File: spec/models/navigation_spec.rb [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+    it "handles '&'" do
+      @title = 'Fines & Fees'
+      expect(subject).to eq('fines-and-fees')
+    end
+    it "handles '\'" do
+      @title = 'Document Delivery / Scans'
+      expect(subject).to eq('document-delivery-or-scans')
+    end
+    it "turns 'Account overview' into '/'" do
+      @title = 'Account overview'
```
Comment (erinesullivan) [inline spec/models/navigation_spec.rb]:
```suggestion
      @title = 'Account Overview'
```

File: spec/models/navigation_spec.rb [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+      expect(subject).to eq(true)
+    end
+    it "returns false if slug of current path does not match page or child page" do
+      @args[:title] = "Some Other Top Level Page" #title
+      @args[:children] = [{"title" => 'Some Other Child'}] #children
+      @args[:current_path] = '/top-level-path/next-level-path' #current path
+      expect(subject).to eq(false)
+    end
+    it "is true for 'Account overview' if path is '/'" do
+      @args[:title] = "Account overview"
```
Comment (erinesullivan) [inline spec/models/navigation_spec.rb]:
```suggestion
      @args[:title] = "Account Overview"
```

File: spec/models/navigation_spec.rb [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+      @args[:current_path] = "/"
+      expect(subject).to eq(true)
+    end
+    it "is false for 'Account overview' if path is not '/'" do
+      @args[:title] = "Account overview"
```
Comment (erinesullivan) [inline spec/models/navigation_spec.rb]:
```suggestion
      @args[:title] = "Account Overview"
```

File: spec/models/navigation_spec.rb [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+    end
+    it "handles '&'" do
+      @title = 'Fines & Fees'
+      expect(subject).to eq('fines-and-fees')
+    end
+    it "handles '\'" do
+      @title = 'Document Delivery / Scans'
+      expect(subject).to eq('document-delivery-or-scans')
+    end
+    it "turns 'Account overview' into '/'" do
```
Comment (erinesullivan) [inline spec/models/navigation_spec.rb]:
```suggestion
    it "turns 'Account Overview' into '/'" do
```

File: spec/models/navigation_spec.rb [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+      @args[:current_path] = '/top-level-path/next-level-path' #current path
+      expect(subject).to eq(true)
+    end
+    it "returns false if slug of current path does not match page or child page" do
+      @args[:title] = "Some Other Top Level Page" #title
+      @args[:children] = [{"title" => 'Some Other Child'}] #children
+      @args[:current_path] = '/top-level-path/next-level-path' #current path
+      expect(subject).to eq(false)
+    end
+    it "is true for 'Account overview' if path is '/'" do
```
Comment (erinesullivan) [inline spec/models/navigation_spec.rb]:
```suggestion
    it "is true for 'Account Overview' if path is '/'" do
```

File: spec/models/navigation_spec.rb [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+      @args[:title] = "Account overview"
+      @args[:current_path] = "/"
+      expect(subject).to eq(true)
+    end
+    it "is false for 'Account overview' if path is not '/'" do
```
Comment (erinesullivan) [inline spec/models/navigation_spec.rb]:
```suggestion
    it "is false for 'Account Overview' if path is not '/'" do
```

**Review by erinesullivan — CHANGES_REQUESTED** [review]
More nit-picky titling. Sorry!

File: config/navigation.json [hunk 13]
```diff
@@ -0,0 +1,74 @@
+[
+  { 
+    "title": "Account Overview"
```
Comment (erinesullivan) [inline config/navigation.json]:
```suggestion
    "title": "Account Overview",
    "icon_name": "home"
```

**Review by erinesullivan — CHANGES_REQUESTED** [review]
Missing an icon.

**Review by erinesullivan — APPROVED** [review]
Hooray for simplifying!

Code is clean, nothing seems broken in the site, and tests pass. 👍🏻

**Pull request merged**